File: agriculture-field-service/src/main/java/com/omstu/agriculturefield/field/service/SoilHorizonService.java
package com.omstu.agriculturefield.field.service;

import com.omstu.agriculturefield.field.dto.SoilHorizonRequest;
import com.omstu.agriculturefield.field.dto.SoilHorizonResponse;

import java.util.List;

public interface SoilHorizonService {
    List<SoilHorizonResponse> getByFieldId(Long fieldId);
    SoilHorizonResponse getById(Long id);
    SoilHorizonResponse create(SoilHorizonRequest request);
    SoilHorizonResponse update(Long id, SoilHorizonRequest request);
    void delete(Long id);
}

File: frontend/src/services/soilHorizonService.ts
import { SoilHorizon, SoilHorizonRequest } from '../types/AgronomicTypes'
import { api } from './api'

export const soilHorizonService = {
  async getByField(fieldId: number): Promise<SoilHorizon[]> {
    const response = await api.get<SoilHorizon[]>('/soil-horizons', { params: { fieldId } })
    return response.data
  },

  async getById(id: number): Promise<SoilHorizon> {
    const response = await api.get<SoilHorizon>(`/soil-horizons/${id}`)
    return response.data
  },

  async create(data: SoilHorizonRequest): Promise<SoilHorizon> {
    const response = await api.post<SoilHorizon>('/soil-horizons', data)
    return response.data
  },

  async update(id: number, data: SoilHorizonRequest): Promise<SoilHorizon> {
    const response = await api.put<SoilHorizon>(`/soil-horizons/${id}`, data)
    return response.data
  },

  async delete(id: number): Promise<void> {
    await api.delete(`/soil-horizons/${id}`)
  },
}
